src/CoreWCF.RabbitMQ.Client/src/CoreWCF/ServiceModel/Channels/RabbitMqBinding.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Net;
+using CoreWCF.Channels;
+using RabbitMQ.Client;
+using BinaryMessageEncodingBindingElement = System.ServiceModel.Channels.BinaryMessageEncodingBindingElement;
+using Binding = System.ServiceModel.Channels.Binding;
+using BindingElementCollection = System.ServiceModel.Channels.BindingElementCollection;
+using TextMessageEncodingBindingElement = System.ServiceModel.Channels.TextMessageEncodingBindingElement;
+
+namespace CoreWCF.ServiceModel.Channels
+{
+    public class RabbitMqBinding : Binding
+    {
+        private RabbitMqTransportBindingElement _transport;
+        private TextMessageEncodingBindingElement _textMessageEncodingBindingElement;
+        private BinaryMessageEncodingBindingElement _binaryMessageEncodingBindingElement;
+
+        public RabbitMqBinding(string uri) : this(new Uri(uri))
+        { }
+
+        public RabbitMqBinding(Uri uri)
+        {
+            _textMessageEncodingBindingElement = new TextMessageEncodingBindingElement();
+            _binaryMessageEncodingBindingElement = new BinaryMessageEncodingBindingElement();
+
+            var sslOption = new SslOption();
+            var virtualHost = ConnectionFactory.DefaultVHost;
+            var credentials = new NetworkCredential(ConnectionFactory.DefaultUser, ConnectionFactory.DefaultPass);
+
+            _transport = new RabbitMqTransportBindingElement
+            {
+                BaseAddress = uri,
+                Credentials = credentials,
+                SslOption = sslOption,
+                VirtualHost = virtualHost
+            };
+        }
+
+        /// <summary>
+        /// Specifies the maximum encoded message size
+        /// </summary>
+        public long MaxMessageSize
+        {
+            get => _transport.MaxReceivedMessageSize;
+            set => _transport.MaxReceivedMessageSize = value;
+        }
+
+        /// <summary>
+        /// Specifies the version of the AMQP protocol that should be used to communicate with the broker
+        /// </summary>
+        public IProtocol BrokerProtocol
+        {
+            get => _transport.BrokerProtocol;
+            set => _transport.BrokerProtocol = value;
+        }
+
+        public Uri BaseAddress
+        {
+            get => _transport.BaseAddress;
+            set => _transport.BaseAddress = value;
+        }
+
+        public SslOption SslOption
+        {
+            get => _transport.SslOption;
+            set => _transport.SslOption = value;
+        }
+
+        public string VirtualHost
+        {
+            get => _transport.VirtualHost;
+            set => _transport.VirtualHost = value;
+        }
+
+        /// <summary>
+        /// Credentials to access the RabbitMQ host
+        /// </summary>
+        public ICredentials Credentials
+        {
```

Comment:
Credentials should probably come form the ClientCredentials. In the relevant place in the ChannelFactory code I'll provide a link to some sample code showing how to pull the credentials from the ClientCredentials class